Couldn't place player in world- Invalid player data

This issue is not solved in a development build

**Describe the bug**
Unclear the cause of this issue, when attempting to enter our lobby sometimes it kicks you with the message "Invalid player data" if you attempt to rejoin 1-2 more times it let's you in without an issue

Error on console:
```
[22:48:04 INFO]: UUID of player AyalX is 211c5729-e7a3-4054-af6f-e75b955a126b
[22:48:04 ERROR]: Couldn't place player in world
net.minecraft.ReportedException: Sending packet
        at net.minecraft.server.network.ServerGamePacketListenerImpl.send(ServerGamePacketListenerImpl.java:2260) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.send(ServerGamePacketListenerImpl.java:2238) ~[?:?]
        at net.minecraft.server.players.PlayerList.sendPlayerPermissionLevel(PlayerList.java:1156) ~[purpur-1.19.4.jar:git-Purpur-1985]
        at net.minecraft.server.players.PlayerList.sendPlayerPermissionLevel(PlayerList.java:1140) ~[purpur-1.19.4.jar:git-Purpur-1985]
        at net.minecraft.server.players.PlayerList.sendPlayerPermissionLevel(PlayerList.java:985) ~[purpur-1.19.4.jar:git-Purpur-1985]
        at net.minecraft.server.players.PlayerList.placeNewPlayer(PlayerList.java:290) ~[purpur-1.19.4.jar:git-Purpur-1985]
        at net.minecraft.server.network.ServerLoginPacketListenerImpl.placeNewPlayer(ServerLoginPacketListenerImpl.java:202) ~[?:?]
        at net.minecraft.server.network.ServerLoginPacketListenerImpl.handleAcceptedLogin(ServerLoginPacketListenerImpl.java:183) ~[?:?]
        at net.minecraft.server.network.ServerLoginPacketListenerImpl.tick(ServerLoginPacketListenerImpl.java:85) ~[?:?]
        at net.minecraft.network.Connection.tick(Connection.java:604) ~[?:?]
        at net.minecraft.server.network.ServerConnectionListener.tick(ServerConnectionListener.java:234) ~[?:?]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1605) ~[purpur-1.19.4.jar:git-Purpur-1985]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:488) ~[purpur-1.19.4.jar:git-Purpur-1985]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1424) ~[purpur-1.19.4.jar:git-Purpur-1985]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1195) ~[purpur-1.19.4.jar:git-Purpur-1985]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:325) ~[purpur-1.19.4.jar:git-Purpur-1985]
        at java.lang.Thread.run(Thread.java:840) ~[?:?]
Caused by: java.lang.NullPointerException: type cannot be null
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:907) ~[guava-31.1-jre.jar:?]
        at com.comphenix.protocol.injector.StructureCache.getStructure(StructureCache.java:138) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.events.PacketContainer.<init>(PacketContainer.java:126) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.injector.netty.manager.NetworkManagerInjector.onPacketSending(NetworkManagerInjector.java:96) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.injector.netty.channel.NettyChannelInjector.processOutbound(NettyChannelInjector.java:553) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.injector.netty.channel.NettyChannelInjector$2.doProxyRunnable(NettyChannelInjector.java:456) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.injector.netty.channel.NettyEventLoopProxy.proxyRunnable(NettyEventLoopProxy.java:43) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.injector.netty.channel.NettyEventLoopProxy.execute(NettyEventLoopProxy.java:252) ~[ProtocolLib.jar:?]
        at net.minecraft.network.Connection.sendPacket(Connection.java:442) ~[?:?]
        at net.minecraft.network.Connection.send(Connection.java:385) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.send(ServerGamePacketListenerImpl.java:2252) ~[?:?]
        ... 16 more
[22:48:04 INFO]: AyalX lost connection: Invalid player data
```

**Version Info**
5.1.1 SNAPSHOT 673
[dump-2023-11-22_20.47.16.txt](https://github.com/dmulloy2/ProtocolLib/files/13444199/dump-2023-11-22_20.47.16.txt)

**Additional context**
git-Purpur-1985 MC 1.19.4 (Latest currently)

Wait i also getting this error in my bungee logs

` A decoder exception has been thrown:
io.netty.handler.codec.DecoderException: xyz.yooniks.escanorproxy.utils.FastOverflowPacketException: Cannot receive string longer than 16 (got 17 characters)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:98) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:333) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:454) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:800) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:509) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:407) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[aegis.jar:git:Aegis-Bootstrap:1.20-R0.1-SNAPSHOT:95137c8:unknown]
	at java.lang.Thread.run(Thread.java:842) ~[?:?]
Caused by: xyz.yooniks.escanorproxy.utils.FastOverflowPacketException: Cannot receive string longer than 16 (got 17 characters)`

> This problem could be related to a 3rd party plugin sending a packet in the wrong state (most likely a PLAY packet during CONFIGURATION), maybe also related #2644.
> 
> I improved the error handling to make the log output more informative, as I cannot reproduce the problem. Please use this version of ProtocolLib on your server: https://github.com/lukalt/ProtocolLib/actions/runs/7400452537/artifacts/1145817298 and share the server log containing the error message.
> 
> I am not quite sure if all the comments in this issue refer to exactly the same issue

Is it referring to this log? I'm using 1.20.3 for testing
```YAML
[03:47:11 ERROR]: Couldn't place player in world
net.minecraft.ReportedException: Sending packet
        at net.minecraft.server.network.ServerGamePacketListenerImpl.send(ServerGamePacketListenerImpl.java:2131) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.send(ServerGamePacketListenerImpl.java:2109) ~[?:?]
        at net.minecraft.server.players.PlayerList.placeNewPlayer(PlayerList.java:280) ~[purpur-1.18.2.jar:git-Purpur-1632]
        at net.minecraft.server.network.ServerLoginPacketListenerImpl.placeNewPlayer(ServerLoginPacketListenerImpl.java:215) ~[?:?]
        at net.minecraft.server.network.ServerLoginPacketListenerImpl.handleAcceptedLogin(ServerLoginPacketListenerImpl.java:196) ~[?:?]
        at net.minecraft.server.network.ServerLoginPacketListenerImpl.tick(ServerLoginPacketListenerImpl.java:84) ~[?:?]
        at net.minecraft.network.Connection.tick(Connection.java:567) ~[?:?]
        at net.minecraft.server.network.ServerConnectionListener.tick(ServerConnectionListener.java:231) ~[?:?]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1651) ~[purpur-1.18.2.jar:git-Purpur-1632]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:522) ~[purpur-1.18.2.jar:git-Purpur-1632]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1483) ~[purpur-1.18.2.jar:git-Purpur-1632]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1247) ~[purpur-1.18.2.jar:git-Purpur-1632]
        at net.minecraft.server.MinecraftServer.lambda$spin$1(MinecraftServer.java:321) ~[purpur-1.18.2.jar:git-Purpur-1632]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.IllegalStateException: Unable to retrieve packet type for net.minecraft.network.protocol.game.PacketPlayOutRecipeUpdate
 in protocol state LOGIN
        at com.comphenix.protocol.injector.netty.manager.NetworkManagerInjector.onPacketSending(NetworkManagerInjector.java:98) ~[ProtocolLi
b.jar:?]
        at com.comphenix.protocol.injector.netty.channel.NettyChannelInjector.processOutbound(NettyChannelInjector.java:553) ~[ProtocolLib.j
ar:?]
        at com.comphenix.protocol.injector.netty.channel.NettyChannelInjector$2.doProxyRunnable(NettyChannelInjector.java:456) ~[ProtocolLib
.jar:?]
        at com.comphenix.protocol.injector.netty.channel.NettyEventLoopProxy.proxyRunnable(NettyEventLoopProxy.java:43) ~[ProtocolLib.jar:?]
        at com.comphenix.protocol.injector.netty.channel.NettyEventLoopProxy.execute(NettyEventLoopProxy.java:252) ~[ProtocolLib.jar:?]
        at net.minecraft.network.Connection.writePacket(Connection.java:435) ~[?:?]
        at net.minecraft.network.Connection.send(Connection.java:382) ~[?:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.send(ServerGamePacketListenerImpl.java:2123) ~[?:?]
        ... 13 more
[03:47:11 INFO]: admiadc lost connection: Invalid player data
> pl
[03:51:52 INFO]: Plugins (27): Bank, CMI, CMILib, Core, DecentHolograms, EternalTags, FluidLimit*, Ip2region, ItemsAdder, LibertyBans, LiteS
ignIn, LoneLibs, LuckPerms, PixelVip, PlaceholderAPI, PlugManX (PlugMan), ProtocolLib, rustyconnector-paper, SkinsRestorer, SuperLobby, TAB-
Bridge, TrChat, Vault, VelocityReport, ViaVersion, Vulcan, XConomy
```

--- a/src/main/java/com/comphenix/protocol/injector/packet/PacketRegistry.java
+++ b/src/main/java/com/comphenix/protocol/injector/packet/PacketRegistry.java
@@ -712,6 +712,15 @@ public class PacketRegistry {
 			return PacketType.Play.Server.BUNDLE;
 		}
 
+		/*
+		 * Reverts https://github.com/dmulloy2/ProtocolLib/pull/2568 for server versions 1.8 to 1.20.1.
+		 *
+		 * Since packet classes are not shared for these versions,
+		 * the protocol state is not needed to determine the packet type from class.
+		 */
+		if (!MinecraftVersion.CONFIG_PHASE_PROTOCOL_UPDATE.atOrAbove()) {
+			return getPacketType(packet);
+		}
 		Map<Class<?>, PacketType> classToTypesForProtocol = REGISTER.protocolClassToType.get(protocol);
 		return classToTypesForProtocol == null ? null : classToTypesForProtocol.get(packet);
 	}

